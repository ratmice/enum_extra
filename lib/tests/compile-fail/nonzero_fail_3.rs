use enum_extra_derive::NonZeroRepr;
use enum_extra::NonZeroRepr;
use strum_macros::EnumMetadata;
use strum::EnumMetadata;

#[repr(C)]
union Error {
  non_error: NonError,
  concrete_error: ConcreteError,
  error: isize
}

#[derive(Copy, Clone)]
#[repr(isize)]
enum NonError {
  Ok = -1,
}


#[derive(Copy, Clone)]
#[derive(EnumMetadata, NonZeroRepr)]
#[repr(isize)]
enum ConcreteError {
  Foo = NonError::Ok as isize + 1,
  Bar,
}

fn main() {

}
