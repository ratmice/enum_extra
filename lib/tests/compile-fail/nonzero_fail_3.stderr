error[E0080]: evaluation of constant value failed
  --> tests/compile-fail/nonzero_fail_3.rs:21:24
   |
21 | #[derive(EnumMetadata, NonZeroRepr)]
   |                        ^^^^^^^^^^^ the evaluated program panicked at 'Expected non-zero discriminant expression', $DIR/tests/compile-fail/nonzero_fail_3.rs:21:24
   |
   = note: this error originates in the macro `$crate::panic::panic_2015` (in Nightly builds, run with -Z macro-backtrace for more info)
